File: BinaryTree/BinaryTree.js
class BinaryTree {
  constructor (root) {
    this.root = root
  }

  traverseDF (node) {
    console.log(`TRAVERSING DF ${node.data}`)
    node.children.forEach(childNode => {
      this.traverseDF(childNode)
    })
  }

  traverseBF (node) {
    let list = []
    while (node) {
      node.children.forEach(childNode => {
        list.push(childNode)
      })

      console.log(`TRAVERSING BF ${node.data}`)

      node = list.splice(0, 1)[0]
    }
  }
}

module.exports = BinaryTree

File: py/binarytree/binary_tree.py
class BinaryTree(object):
    def __init__(self, root):
        self.root = root

    def traverse_df(self, node):
        print(f'TRAVERSING DF {node.data}')
        for child_node in node.children:
            self.traverse_df(child_node)

    def traverse_bf(self, node):
        lst = []
        while node:
            for child_node in node.children:
                lst.append(child_node)

            print(f'TRAVERSING BF {node.data}')

            node = lst.pop(0) if len(lst) > 0 else None
